server/src/routes/serializers/TrailSerializer.js

```diff
@@ -0,0 +1,18 @@
+import ReviewSerializer from "./ReviewSerializer.js"
+
+class TrailSerializer {
+  static async getSummary(trail) {
+    const allowedAttributes = ["id", "name", "length", "description", "location", "estimateTime"]
+
+    let serializedTrail = {}
+    for (const attribute of allowedAttributes) {
+      serializedTrail[attribute] = trail[attribute]
+    }
+    const relatedReviews = await trail.$relatedQuery("reviews")
+    const serializedTrails = relatedReviews.map((review) => ReviewSerializer.getSummary(review))
```

Comment:
This is really the `serailzedReviews` we will be returning from the `.map()` function.  I would update the variable name here and on line 13

Reply:
Since `ReviewSerializer.getSummary()` is an async function we will need to make use of `Promise.all` and call

```js
const serializedTrails = await Promise.all(
  relatedReviews.map((review) => ReviewSerializer.getSummary(review))
)
```

The other alternative would be to use a `for` loop with `return await` syntax on each iteration